raiden/tests/unit/test_channelstate.py

```diff
@@ -1910,6 +1927,257 @@ def test_receive_withdraw_confirmation():
+        pseudo_random_generator=pseudo_random_generator,
+    )
+
+    assert (
+        search_for_item(
+            iteration.events,
+            SendWithdrawExpired,
+            {
+                "total_withdraw": total_withdraw,
+                "participant": channel_state.our_state.address,
+                "recipient": channel_state.partner_state.address,
+                "nonce": 1,
+            },
+        )
+        is not None
+    )
+
+
+def test_node_handles_received_withdraw_expiry():
+    pseudo_random_generator = random.Random()
+
+    our_model1, _ = create_model(balance=70)
+    partner_model1, privkey2 = create_model(balance=100)
+    signer = LocalSigner(privkey2)
+    channel_state = create_channel_from_models(our_model1, partner_model1, privkey2)
+    block_hash = make_block_hash()
+
+    total_withdraw = 50
+    expiration_block_number = 10
+    expiration_threshold = channel.get_receiver_expiration_threshold(expiration_block_number)
+
+    channel_state.partner_state.total_withdraw = total_withdraw
+    channel_state.partner_state.withdraws.append(
+        WithdrawState(total_withdraw=total_withdraw, expiration=expiration_block_number, nonce=1)
+    )
+
+    packed = pack_withdraw(
+        canonical_identifier=channel_state.canonical_identifier,
+        # pylint: disable=no-member
+        participant=channel_state.partner_state.address,
+        # pylint: enable=no-member
+        total_withdraw=total_withdraw,
+    )
+    partner_signature = signer.sign(packed)
+
+    # Test a withdraw that has not expired yet
+    receive_withdraw_expired = ReceiveWithdrawExpired(
+        message_identifier=message_identifier_from_prng(pseudo_random_generator),
+        canonical_identifier=channel_state.canonical_identifier,
+        total_withdraw=total_withdraw,
+        signature=partner_signature,
+        # pylint: disable=no-member
+        sender=channel_state.partner_state.address,
+        # pylint: enable=no-member
+        nonce=1,
+    )
+
+    iteration = channel.state_transition(
+        channel_state=channel_state,
+        state_change=receive_withdraw_expired,
+        block_number=expiration_threshold - 1,
+        block_hash=block_hash,
+        pseudo_random_generator=pseudo_random_generator,
+    )
+
+    assert (
+        search_for_item(
+            iteration.events,
+            EventInvalidReceivedWithdrawExpired,
+            {
+                "attempted_withdraw": total_withdraw,
+                "reason": f"Invalid WithdrawExpired of {total_withdraw}",
+            },
+        )
+        is not None
+    )
+
+    # Invalid signature
+    receive_withdraw_expired = ReceiveWithdrawExpired(
+        message_identifier=message_identifier_from_prng(pseudo_random_generator),
+        canonical_identifier=channel_state.canonical_identifier,
+        total_withdraw=total_withdraw,
+        signature=make_32bytes(),
+        # pylint: disable=no-member
+        sender=channel_state.partner_state.address,
```

Comment:
weird, any ideas why this failed to see the field? Is it complaining about `partner_state` or `address`?

Reply:
I checked, it seems like the current version of pylint that we use doesn't detect this as a no-member error. Will do in a follow up PR